src/ledger/OfferFrame.cpp

```diff
@@ -120,7 +120,16 @@ OfferFrame::getFlags() const
 bool
 OfferFrame::isValid(OfferEntry const& oe)
 {
-    return isAssetValid(oe.buying) && isAssetValid(oe.selling);
+    bool res = isAssetValid(oe.buying);
+    // to avoid SQL problems (not a protocol limitation)
+    res = res && (oe.offerID <= INT64_MAX);
```

Comment:
If this isn't a protocol limitation, should we actually be checking this here? We aren't performing analogous checks in other *Frame::isValid functions for values that are BIGINT.

Reply:
Yes we should check - I've removed the comment that is confusing:
At the protocol level, we want those to be unsigned - yet most current implementations won't be able to deal with values that are bigger than max int64; this is not a problem for now as the sun should have exploded by the time we reach those type of numbers.

I've done an extra pass on the isValid functions to address this widely.